### PR TITLE
 Android camera event fixes + misc cleanup

### LINKED ISSUE #323: onRegionIsChanging is not working on Android.
The onRegionIsChanging event is not working for me in android. I am trying to call onRegionIsChanging to get the heading value but it gets called only when the compass heading is zero. For ios, there is no issue.
```
 <MapboxGL.MapView
          onRegionIsChanging={this.onRegionIsChanging}
          onRegionDidChange={this.onRegionDidChange}
          ref={c => (this._map = c)}
          onPress={this.onPress}
          style={{ flex: 1 }}
        >
          <MapboxGL.Camera
            zoomLevel={9}
            centerCoordinate={[-73.970895, 40.723279]}
          />
        </MapboxGL.MapView>

 onRegionIsChanging() {
    const map = await this._map;
    console.log(
      "onRegionIsChanging event",
      map.state.region.properties.heading
    );
  }
```

```
   "@react-native-mapbox-gl/maps": "^7.0.1",
   "react-native": "^0.59.9"
```

I am trying to get heading value from onRegionIsChanging event (event.properties.heading ).
Is there a way to get the heading value as the user rotates/moves the chart?

**To Reproduce**
checkout my [repo](https://github.com/irfanshaikh138/mapboxgl-test) in that I have used the GetZoom example(search for GetZoom.js).It also presents in the mapbox gl example code.

### PR DESCRIPTION
**Pull request by kristfal**
Fixes #323.

### CODE REVIEW

File: javascript/utils/geoUtils.js [hunk 1]
```diff
@@ -28,18 +28,18 @@ export function makeFeatureCollection(features = []) {
   return featureCollection(features);
 }
 
-export function addToFeatureCollection(featureCollection, feature) {
-  const shallowFeatureCollection = Object.assign({}, featureCollection);
-  shallowFeatureCollection.features.push(feature);
+export function addToFeatureCollection(newFeatureCollection, newFeature) {
+  const shallowFeatureCollection = Object.assign({}, newFeatureCollection);
+  shallowFeatureCollection.features.push(newFeature);
   return featureCollection;
```
Comment (mfazekas) [inline javascript/utils/geoUtils.js]:
```suggestion
  return newFeatureCollection;
```

BTW i'm not sure the idea behind `Object.assign({},` eg why we don't just do

```js
newFeatureCollection.features.push(newFeature);
return newFeatureCollection;
```

File: javascript/components/Camera.js [hunk 2]
```diff
@@ -90,13 +89,10 @@ class Camera extends React.Component {
     followPitch: PropTypes.number,
     followHeading: PropTypes.number,
 
-    // manual update
```
[the remark this answers is not in the document]
Comment (kristfal) [reply javascript/components/Camera.js]: Ah, didn't know. Will revert.

File: javascript/utils/AnimatedPoint.js [hunk 3]
```diff
@@ -52,7 +52,9 @@ export class AnimatedPoint extends AnimatedWithChildren {
   }
 
   addListener(cb) {
-    const id = `${String(uniqueID++)}-${String(Date.now())}`;
+    uniqueID += uniqueID;
```
[the remark this answers is not in the document]
Comment (kristfal) [reply javascript/utils/AnimatedPoint.js]: Yeh, thanks

File: javascript/utils/geoUtils.js (hunk 1 above)
Comment (kristfal) [reply javascript/utils/geoUtils.js]: Thanks, will fix. Reason for Object.assign is to not mutate the input object, which is bad practice.

**Comment by kristfal** [comment]
FYI, the actual fixes are here: https://github.com/react-native-mapbox-gl/maps/pull/367/files#diff-83bd90811e16ee14c9357ca5dd70f50cR412

Yes, will split next time.

File: javascript/index.js [hunk 4]
```diff
@@ -27,14 +22,10 @@ import CircleLayer from './components/CircleLayer';
...
 import BackgroundLayer from './components/BackgroundLayer';
-
-// modules
 import locationManager from './modules/location/locationManager';
 import offlineManager from './modules/offline/offlineManager';
 import snapshotManager from './modules/snapshot/snapshotManager';
-
-// helpers
-import AnimatedPoint from './utils/AnimatedPoint';
+import AnimatedMapPoint from './utils/AnimatedPoint';
```
[the remark this answers is not in the document]
Comment (kristfal) [reply javascript/index.js]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-named-as-default.md

We could drop it

File: javascript/utils/geoUtils.js (hunk 1 above)
Comment (mfazekas) [reply javascript/utils/geoUtils.js]: It mutates `newFeatureCollection.features` array, but not `newFeatureCollection` object itself, so i don't think assign is needed.

**Pull request merged**